netebpfext/net_ebpf_ext.c

```diff
@@ -156,6 +189,72 @@ static HANDLE _fwp_engine_handle;
 // WFP component management related utility functions.
 //
 
+net_ebpf_extension_hook_id_t
+net_ebpf_extension_get_hook_id_from_wfp_layer_id(uint16_t wfp_layer_id)
+{
+    net_ebpf_extension_hook_id_t hook_id = 0;
+    ASSERT(
+        (wfp_layer_id == FWPS_LAYER_OUTBOUND_MAC_FRAME_NATIVE) ||
+        (wfp_layer_id == FWPS_LAYER_INBOUND_MAC_FRAME_NATIVE) ||
+        (wfp_layer_id == FWPS_LAYER_ALE_RESOURCE_ASSIGNMENT_V4) ||
+        (wfp_layer_id == FWPS_LAYER_ALE_RESOURCE_ASSIGNMENT_V6) ||
+        (wfp_layer_id == FWPS_LAYER_ALE_RESOURCE_RELEASE_V4) || (wfp_layer_id == FWPS_LAYER_ALE_RESOURCE_RELEASE_V6) ||
+        (wfp_layer_id == FWPS_LAYER_ALE_AUTH_CONNECT_V4) || (wfp_layer_id == FWPS_LAYER_ALE_AUTH_CONNECT_V6) ||
+        (wfp_layer_id == FWPS_LAYER_ALE_AUTH_RECV_ACCEPT_V4) || (wfp_layer_id == FWPS_LAYER_ALE_AUTH_RECV_ACCEPT_V6) ||
+        (wfp_layer_id == FWPS_LAYER_ALE_FLOW_ESTABLISHED_V4) || (wfp_layer_id == FWPS_LAYER_ALE_FLOW_ESTABLISHED_V6));
```

Comment:
The reason why I added this seemingly-ridiculous assert is because one of the previous feedback was to avoid ASSERT(FALSE). I will gladly change this.